OrthoEvol/Tools/ftp/ncbiftp.py

```diff
@@ -355,21 +382,29 @@ def getblastdb(self, database_name, download_path, extract=True):
                                   (len(absentfiles), absentfiles))
             # Download the files using multiprocessing
             download_time_secs = time()
-            with ThreadPool(1) as download_pool:
-                with tqdm(total=len(self.files2download)) as pbar:
-                    for i, _ in tqdm(enumerate(download_pool.imap(self.download_file, self.files2download))):
-                        pbar.update()
-                minutes = round(((time() - download_time_secs) / 60), 2)
-            self.ncbiftp_log.info("Took %s minutes to download the files.\n" %
-                                  minutes)
+
+            try:
+                with ThreadPool(self.cpus) as download_pool:
+                    download_pool.map(self.download_file, self.files2download)
+                    minutes = round(((time() - download_time_secs) / 60), 2)
+                self.ncbiftp_log.info("Took %s minutes to download the files.\n" %
+                                      minutes)
+
+            except RuntimeError as err:
+                self.ncbiftp_log.error(err)
+                # Try to download again
+                self.ncbiftp_log.warning('Attempting to download again.')
+                with ThreadPool(self.cpus) as download_pool:
+                    download_pool.map(self.download_file, self.files2download)
+                    minutes = round(((time() - download_time_secs) / 60), 2)
+                self.ncbiftp_log.info("Took %s minutes to download the files.\n" %
+                                      minutes)
 
         if extract:
             self.ncbiftp_log.info('Now it\'s time to extract files.')
             extract_time_secs = time()
-            with ThreadPool(3) as extract_pool:
-                with tqdm(total=len(self.files2download)) as pbar:
-                    for _ in tqdm(enumerate(extract_pool.imap(self.extract_file, self.files2download))):
```

Comment:
What happened to the progress bar?  Did it stop working?

Reply:
Honestly, I'm not sure why this was removed. I don't remember removing it so I'll look over this.

Reply:
Probably a poorly resolved conflict or something similar is my guess.

Reply:
Yeah, I checked https://github.com/datasnakes/OrthoEvolution/commit/94a48961eb957d537074fc80fb5bf396c3c58175#diff-976275deb06d5bdf9802a051b513221c and found it there. I'll add it back and make sure nothing else was erroneously removed.